acasclient/ddict.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import unicode_literals
+from .validation import validation_result
+
+import logging
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class DDict(object):
+    """The DDict class is meant as a generic interface for any implementation of a Data Dictionary that
+    can be referenced by an ACAS CodeValue.
+    Any classes implementing this interface must implement the update_valid_values() method.
+    The `update_valid_values` method must also be called before calling `check_value`
+    """
+
+    def __init__(self, code_type, code_kind, code_origin):
+        self.code_type = code_type
+        self.code_kind = code_kind
+        self.code_origin = code_origin
+        self.valid_values = None
+
+    def update_valid_values(self, client):
+        raise NotImplementedError()
+
+    @validation_result
+    def check_value(self, value):
+        """Check if the value is within `self.valid_values` for the DDict."""
+        if not self.valid_values:
```

Comment:
Curious: Do we want to show the custom message if `valid_values` is an empty list or only when `valid_values` is `None`? Checking if we should have this as,
```python
if self.valid_values is None:
    ...
```

Reply:
Yeah, this maybe could be better but in the ACASDDict implementation of `update_valid_values` we throw a specific error if no valid values come back (empty list), so I think this will only be hit when someone calls without first calling `update_valid_values`. Going to merge this as-is.